lib/ansible/galaxy/role.py

```diff
@@ -64,7 +65,7 @@ def __init__(self, galaxy, api, name, src=None, version=None, scm=None, path=Non
 
         self.name = name
         self.version = version
-        self.src = src or name
+        self.src = unfrackpath(src or name)
```

Comment:
We could be using unfrackpath on a URL, git repo, or something that provides an absolute path prefixed with file://. This would break those scenarios since we automatically prepend a base dir in unfrackpath.

```
>>> from ansible.utils.path import unfrackpath
>>> unfrackpath('https://github.com/')
'/home/shertel/ansible/https:/github.com'
>>> unfrackpath('file:///path')
'/home/shertel/ansible/file:/path'
>>> unfrackpath('git+file:///path')
'/home/shertel/ansible/git+file:/path'
```

We might be able to move this somewhere that's only supposed to handle directories, or your original `os.path.expanduser()` may be a better choice.

This definitely needs tests.